File: src/main/kotlin/com/sourcegraph/cody/agent/protocol/TestingParams.kt
package com.sourcegraph.cody.agent.protocol

data class TestingParams(
    val selectedText: String? = null,
    val sourceOfTruthDocument: ProtocolTextDocument? = null,
) {
  companion object {
    val doIncludeTestingParam =
        "true".equals(System.getProperty("cody-agent.panic-when-out-of-sync", "false"))
  }
}

File: agent/bindings/csharp/TestingParams.cs
using System.Text.Json.Serialization;

namespace Cody.Core.Agent.Protocol
{
  public class TestingParams
  {
    [JsonProperty(PropertyName = "selectedText")]
    public string SelectedText { get; set; }
    [JsonProperty(PropertyName = "sourceOfTruthDocument")]
    public ProtocolTextDocument SourceOfTruthDocument { get; set; }
  }
}
